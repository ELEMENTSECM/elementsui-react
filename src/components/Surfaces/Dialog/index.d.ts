import * as React from 'react';

export interface DialogDialogContentProps {
	type?: number;
	title?: string;
	subText?: string;
}

export interface DialogModalProps {
	isBlocking?: boolean;
	isDarkOverlay?: boolean;
	containerClassName?: string;
}

export interface DialogProps {
	/**
	 * Dialog visibility state
	 */
	hidden?: boolean;
	/**
	 * Dialog dissmiss callback
	 */
	onDismiss?: (...args: any[]) => any;
	/**
	 * Dialog content props
	 */
	dialogContentProps?: DialogDialogContentProps;
	/**
	 * Dialog modal props (context)
	 */
	modalProps?: DialogModalProps;
	/**
	 * User-defined styling
	 */
	getStyles?: (...args: any[]) => any;
}

declare const Dialog: React.SFC<DialogProps>;

export default Dialog;
