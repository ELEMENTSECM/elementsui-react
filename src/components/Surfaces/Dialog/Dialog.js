import * as React from 'react';
import PropTypes from 'prop-types';
import { Dialog as UIFabDialog } from 'office-ui-fabric-react/lib/Dialog';
import { getStyles } from './Dialog.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** Dialog example */
export function Dialog(props) {
	const { hidden, onDismiss, dialogContentProps, modalProps, className, theme, getStyles } = props;
	const classNames = classNamesFunction()(getStyles, props);

	return (
		<UIFabDialog
			className={classNames.root}
			hidden={hidden}
			onDismiss={onDismiss}
			dialogContentProps={dialogContentProps}
			modalProps={modalProps}>
			{props.children}
		</UIFabDialog>
	);
}

Dialog.propTypes = {
	/** Dialog visibility state */
	hidden: PropTypes.bool,
	/** Dialog dissmiss callback */
	onDismiss: PropTypes.func,
	/** Dialog content props */
	dialogContentProps: PropTypes.shape({
		type: PropTypes.number,
		title: PropTypes.string,
		subText: PropTypes.string
	}),
	/** Dialog modal props (context)*/
	modalProps: PropTypes.shape({
		isBlocking: PropTypes.bool,
		isDarkOverlay: PropTypes.bool,
		containerClassName: PropTypes.string
	}),
	/** User-defined styling */
	getStyles: PropTypes.func
};

export default styled(customizable('Dialog', ['theme'])(Dialog), getStyles);
