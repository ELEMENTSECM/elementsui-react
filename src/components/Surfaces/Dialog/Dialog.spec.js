import React from 'react';
import renderer from 'react-test-renderer';
import Dialog from './Dialog';

describe('Dialog', () => {
	const dialogContentProps = {
		title: 'Dialog header',
		subText: 'Dialog content text with some description of what the dialog reveals.'
	};

	test('displays `Dialog` with dialogContentProps', () => {
		const tree = renderer
			.create(<Dialog htmlId="defaultDialog" dialogContentProps={dialogContentProps} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
