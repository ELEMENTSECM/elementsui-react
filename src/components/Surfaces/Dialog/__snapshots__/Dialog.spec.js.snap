// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog displays \`Dialog\` with dialogContentProps 1`] = `null`;
